From db961f935b0264bb2849d0086e4193efd86290f7 Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Wed, 29 Oct 2014 16:34:37 -0400
Subject: [fs] ext4: get rid of code duplication

Message-id: <1414600485-1929-48-git-send-email-lczerner@redhat.com>
Patchwork-id: 98541
O-Subject: [RHEL 7.1 PATCH v2 47/55] ext4: get rid of code duplication
Bugzilla: 1156096
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: Dmitry Monakhov <dmonakhov@openvz.org>

BZ 1156096
https://bugzilla.redhat.com/show_bug.cgi?id=1156096

BREW 8169722
https://brewweb.devel.redhat.com/taskinfo?taskID=8169722

Upstream commit dfe076c106f63cf6bcd375c56db9c8c89a088dab

 Reviewed-by: Jan Kara <jack@suse.cz>
 Signed-off-by: Dmitry Monakhov <dmonakhov@openvz.org>
 Signed-off-by: Theodore Ts'o <tytso@mit.edu>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/mballoc.c b/fs/ext4/mballoc.c
index 4c2867c..f2dd470 100644
--- a/fs/ext4/mballoc.c
+++ b/fs/ext4/mballoc.c
@@ -3153,9 +3153,8 @@ ext4_mb_normalize_request(struct ext4_allocation_context *ac,
     "start %lu, size %lu, fe_logical %lu",
     (unsigned long) start, (unsigned long) size,
     (unsigned long) ac->ac_o_ex.fe_logical);
+  BUG();
  }
- BUG_ON(start + size <= ac->ac_o_ex.fe_logical &&
-   start > ac->ac_o_ex.fe_logical);
  BUG_ON(size <= 0 || size > EXT4_CLUSTERS_PER_GROUP(ac->ac_sb));
 
  /* now prepare goal request */
-- 
1.7.1